src/bin/options.ts

```diff
@@ -0,0 +1,54 @@
+import commander from 'commander';
+import config from '../config';
+
+function parseNumber(v: string): number {
+  const num = parseInt(v);
+  if (isNaN(num)) {
+    throw new commander.InvalidOptionArgumentError(`${v} is not a number`);
+  }
+  return num;
+}
+
+const clientHost = new commander.Option(
+  '-ch, --client-host <address>',
+  'Client Host Address'
+)
+  .env('PK_CLIENT_HOST')
+  .default(config.defaults.clientHost);
+
+const clientPort = new commander.Option(
+  '-cp, --client-port <port>',
+  'Client Port'
+)
+  .argParser(parseNumber)
+  .env('PK_CLIENT_PORT')
+  .default(config.defaults.clientPort);
+
+const ingressHost = new commander.Option(
+  '-ih, --ingress-host <address...>',
+  'ReverseProxy ingress host'
+)
+  .env('PK_INGRESS_HOST')
+  .default(config.defaults.ingressHost);
+
+function hostParser(input) {
+  if (typeof input == 'string') return input;
+  if (input instanceof Array) return input[0];
+  return config.defaults.ingressHost;
+}
```

Comment:
This should be changed to be more inline with how `parseNumber` is done:

```ts
function parseNumber(v: string): number {
  const num = parseInt(v);
  if (isNaN(num)) {
    throw new commander.InvalidOptionArgumentError(`${v} is not a number`);
  }
  return num;
}
```

And how it's integrated directly into the command, rather than separately:

```ts
const ingressHost = new commander.Option(
  '-ih, --ingress-host <address...>',
  'ReverseProxy ingress host'
)
  .env('PK_INGRESS_HOST')
  .default(config.defaults.ingressHost);
  // no .argParser here?

const clientPort = new commander.Option(
  '-cp, --client-port <port>',
  'Client Port'
)
  .argParser(parseNumber)
  .env('PK_CLIENT_PORT')
  .default(config.defaults.clientPort);
```

i.e. some kind of validation logic integrated directly into the parsing function, and throws an error if the host is not valid.

Reply:
Oh, I see that `hostParser` is used such that we can specify multiple `--ingress-host` parameters.

Reply:
In any case, we should have a `parseHost` function that does the same as `parseNumber`, performing some validation logic and throwing an exception if it's not a valid host (I seem to recall there already being utilities in the `network` domain for this - can probably piggyback off those) and apply this to both `ingressHost` and `clientHost` (like how `parseNumber` is used for the ports).

Perhaps we can also change the name of `hostParser` to be `parseMultipleHosts`, or something like that? It wasn't immediately clear to me that this is what it was doing.

Reply:
I'll update it, But it was just there to take the multiple possible hosts for `--ingress-host` and return the first one.

Reply:
I've removed it, it's not actually needed.